canary/generator/download/download_whole.py

```diff
@@ -0,0 +1,46 @@
+from collections import defaultdict
+import json
+from json import JSONDecodeError
+import sys
+import os
+
+import requests
+
+from canary.generator.download.query import build_query_string
+
+if __name__ == '__main__':
+    dates_versions_response = requests.get(
+        'https://aggregates.telemetry.mozilla.org/aggregates_by/build_id/channels/nightly/dates/').text
+    dates_versions = json.loads(dates_versions_response)
+    # Dict consisting of pairs version: list of dates
+    dates_for_versions = defaultdict(list)
+    for item in dates_versions:
+        dates_for_versions[item['version']].append(item['date'])
+    versions = sorted(list(dates_for_versions.keys()))
+
+    for version in versions[-int(sys.argv[2]):]:
```

Comment:
Any reason not to use the `argparse` library instead of directly messing with the `sys.argv`? Using the former allows us to specify help lines for the arguments and easily add in validation checks.

Reply:
Let's file an issue in this repo about this: no need to block on it in this PR.

Reply:
hahah, I changed that already :P